connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/ClientE2ETestSuite.scala

```diff
@@ -156,6 +156,19 @@ class ClientE2ETestSuite extends RemoteSparkSession {
     }
   }
 
+  test("write v2") {
+    try {
+      spark.range(3).writeTo("myTableV2").using("parquet").create()
```

Comment:
This API feels wrong, a few errors I get with this API:
```
spark.range(3).writeTo("myTableV2").create() -> Error: should specify source
```

```
spark.range(3).writeTo("myTableV2").using("parquet").createOrReplace() -> Error: dose not support create or select.
```

```
spark.range(3).writeTo("myTableV2").using("parquet").create()
spark.range(3).writeTo("myTableV2").append() -> Error: Cannot write into v1 table
```

Reply:
hmm... when I add `parquet-hadoop` as a test dependency of `connect-client-jvm` module, the test passed ....

Reply:
One problem is that I'm not sure why the test dependency `parquet-hadoop` need to be added to the `connect-client-jvm `

Reply:
After https://github.com/apache/spark/pull/40109, the current behavior of this client is the same as pyspark.

Reply:
For my understanding, the diff between client test and [server test](https://github.com/apache/spark/blob/master/connector/connect/server/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectProtoSuite.scala#L681-L706), it that the config setting:
```
spark.conf.set("spark.sql.catalog.testcat", classOf[InMemoryTableCatalog].getName)
```
changes the current catalog to a V2 catalog.

@cloud-fan Could you explain the diff better for me? I saw you wrote the [SQL test](https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/DataFrameWriterV2Suite.scala#L55) which also have this settings?

CC @hvanhovell